sdk/core/core-amqp/src/util/utils.ts

```diff
@@ -198,12 +200,37 @@ export class Timeout {
 
 /**
  * A wrapper for setTimeout that resolves a promise after t milliseconds.
- * @param {number} t - The number of milliseconds to be delayed.
+ * @param {number} delayInMs - The number of milliseconds to be delayed.
+ * @param {AbortSignalLike} abortSignal - The abortSignal associated with containing operation.
+ * @param {string} abortErrorMsg - The abort error message associated with containing operation.
  * @param {T} value - The value to be resolved with after a timeout of t milliseconds.
  * @returns {Promise<T>} - Resolved promise
  */
-export function delay<T>(t: number, value?: T): Promise<T> {
-  return new Promise((resolve) => setTimeout(() => resolve(value), t));
+export function delay<T>(
+  delayInMs: number,
+  abortSignal?: AbortSignalLike,
+  abortErrorMsg?: string,
+  value?: T
+): Promise<T> {
+  return new Promise((resolve, reject) => {
+    if (abortSignal && abortSignal.aborted) {
+      return reject(
+        new AbortError(abortErrorMsg ? abortErrorMsg : `The delay was cancelled by the user.`)
+      );
+    }
+
+    const timer = setTimeout(() => {
+      resolve(value);
+    }, delayInMs);
+    if (abortSignal) {
+      abortSignal.addEventListener("abort", () => {
```

Comment:
We arent removing the listener after adding it. Please ensure that it gets removed before the promise gets resolved/rejected

Reply:
How come this isn't done at https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/src/util/cancellableDelay.ts ?

Reply:
It should be done there as well unless @chradek has some other code on the outside that takes care of removing the listener.

@chradek Please take a look at the cancellable delay usage in the event processor

Reply:
@ramya-rao-a is right that the listener should be removed here. It wasn't caught in my code review and won't change the _behavior_ of the delay function, but we should clean up event listeners when they are no longer needed.